src/main/java/minicraft/core/Game.java

```diff
@@ -77,6 +94,25 @@ public static void main(String[] args) {
 
 		setDisplay(new TitleDisplay()); // Sets menu to the title screen.
 
+		defaultResourcePack = new VanillaResourcePack();
+		resourcePackManager = new ResourcePackManager();
+		resourcePackManager.addProvider(new DefaultResourcePackProvider(defaultResourcePack));
+		resourcePackManager.findPacks();
+		resourceManager.registerReloader(new SplashManager());
+		// resourceManager.registerReloader(Sound.reloader);
+		// resourceManager.registerReloader(new LocalizationReloader());
+		// resourceManager.registerReloader(new SoundReloader());
+		// resourceManager.registerReloader(new BookReloader());
+		// resourceManager.registerReloader(new TextureReloader());
+		resourcePackManager.setEnabledPacks(Arrays.asList("vanilla"));
+		Game.reloading = true;
+		Game.currentLoader = resourceManager.reload(resourcePackManager.getEnabledPacks(), () -> {
+			Game.reloading = false;
+			Game.currentLoader = null;
+		});
+
+		// TODO: remove reloading from this method so that the reload overlay can be shown >:(
+		// So remove the splash window too because is not needed. When reloading, no screen and level should be render.
```

Comment:
I do not agree with this as the splash screen directly takes the image file from the jar, so it should not be affected.

Reply:
When the splash screen is shown, assets are being loaded. That should be better.